README.md

```diff
@@ -7,15 +7,19 @@ Champion Needed
 ### Why
 
 The behaviour of `Promise.resolve` includes checking for `then` functions
-on whatever you pass it. In most cases this is probably fine but just talk
-with any avid user of JS and they won't have only nice things to say about
-it.
+on whatever you pass it. This allows interop with legacy third-party 
+Promise implementation protocols, and in most cases does not result in any
+issues, but there remain scenarios where it is unwanted.
 
-This is especially true with dynamic import, which will become part of the
-language spec very soon. There has been a lot of discussion around how an
-import declaration and a dynamic import, because of "thenables," may have
-different things passed to a user which won't even be actual namespaces.
-__This is absolutely crazy.__
+This is especially true with dynamic import of module namespace objects,
+which will become part of the language spec very soon. There has been a
+lot of discussion around how an import declaration and a dynamic import
+(such as https://github.com/tc39/proposal-dynamic-import/issues/47,
+https://github.com/tc39/proposal-dynamic-import/issues/48),
+because a namespace might be interpreted as a "thenable" resulting in
+a non-namespace return from `import()` - which may be very unexpected.
```

Comment:
This should probably also discuss how you can already import * as and then await/Promise.resolve that Module Record, but that this will go from super rare to exceedingly common once import() arrives.